src/main/java/frc/robot/subsystems/pivot/PivotKrakenHardware.java

```diff
@@ -0,0 +1,226 @@
+            PivotConstants.kLeaderMotorID, PivotConstants.kFollowerMotorOpposeMasterDirection));
+
+    // Apply configurations
+    motorConfiguration.Slot0.kP = PivotConstants.kPivotGains.kP();
+    motorConfiguration.Slot0.kI = PivotConstants.kPivotGains.kI();
+    motorConfiguration.Slot0.kD = PivotConstants.kPivotGains.kD();
+
+    motorConfiguration.CurrentLimits.SupplyCurrentLimitEnable =
+        configuration.kEnableSupplyCurrentLimit();
+    motorConfiguration.CurrentLimits.SupplyCurrentLimit = configuration.kSupplyCurrentLimitAmps();
+    motorConfiguration.CurrentLimits.StatorCurrentLimitEnable =
+        configuration.kEnableStatorCurrentLimit();
+    motorConfiguration.CurrentLimits.StatorCurrentLimit = configuration.kStatorCurrentLimitAmps();
+    motorConfiguration.MotorOutput.NeutralMode = configuration.kNeutralMode();
+    motorConfiguration.MotorOutput.Inverted =
+        configuration.kInverted()
+            ? InvertedValue.CounterClockwise_Positive
+            : InvertedValue.Clockwise_Positive;
+    motorConfiguration.Voltage.PeakForwardVoltage = configuration.kPeakForwardVoltage();
+    motorConfiguration.Voltage.PeakReverseVoltage = configuration.kPeakReverseVoltage();
+
+    // Setup feedback sensor for position control
+    if (PivotConstants.kUseThroughBore) {
+      throughBoreEncoder = new DutyCycleEncoder(PivotConstants.kThroughBoreEncoderPort);
+      throughBoreEncoder.setDistancePerRotation(PivotConstants.kThroughBoreEncoderRatio);
+      // Reset the position of the mechanism (not necessarily the motor) to the position of the
+      // through bore encoder, which measures the absolute position of the mechanism.
+      kLeadMotor.setPosition(throughBoreEncoder.getAbsolutePosition(), 1.0);
+    }
+    motorConfiguration.Feedback.SensorToMechanismRatio = PivotConstants.kGearRatio;
+    motorConfiguration.Feedback.FeedbackSensorSource = FeedbackSensorSourceValue.RotorSensor;
+
+    kLeadMotor.getConfigurator().apply(motorConfiguration, 1.0);
+
+    // Get status signals from the motor controller
+    internalPositionRotations = kLeadMotor.getPosition();
+    velocityRotationsPerSec = kLeadMotor.getVelocity();
+    appliedVolts = List.of(kLeadMotor.getMotorVoltage(), kFollowMotor.getMotorVoltage());
+    supplyCurrentAmps = List.of(kLeadMotor.getSupplyCurrent(), kFollowMotor.getSupplyCurrent());
+    temperatureCelsius = List.of(kLeadMotor.getDeviceTemp(), kFollowMotor.getDeviceTemp());
+
+    BaseStatusSignal.setUpdateFrequencyForAll(
+        PivotConstants.kStatusSignalUpdateFrequencyHz,
+        internalPositionRotations,
+        velocityRotationsPerSec,
+        appliedVolts.get(0),
+        appliedVolts.get(1),
+        supplyCurrentAmps.get(0),
+        supplyCurrentAmps.get(1),
+        temperatureCelsius.get(0),
+        temperatureCelsius.get(1));
+
+    // Optimize the CANBus utilization by explicitly telling all CAN signals we
+    // are not using to simply not be sent over the CANBus
```

Comment:
You should describe why you are passing 0 and 1, or at least use comments to give an idea of what is being passed.  I assume this is the call?
https://api.ctr-electronics.com/phoenix6/release/java/com/ctre/phoenix6/hardware/ParentDevice.html#optimizeBusUtilization()

Reply:
Since this code is meant to work with an arm design similar to Redux (where there are two motors working together to move a mass) some of the telemetry data will be the same. Status signals are how we get that telemetry data from a TalonFX motor controller - and it's more convenient to store them in objects that can be accessed throughout the class. 

Knowing this, the internal position and velocity of the two motors must be the same (since they are mechanically linked to the arm) however the applied volts, supply current draw, and temperature would be different for the motors. So to log them more conveniently, I store them in a list where index 0 is the leader motor and index 1 is the follower motor. When setting the update frequency at which telemetry to be set, every status signal that I wish to modify must be passed into the method so I call both the leader's (0) and the follower's (1) status signals. I can add documentation if it's not clear in the code that this was just a way of conveniently storing the status signals.